react/src/components/search-app/SearchBarComponent.jsx

```diff
@@ -28,12 +28,12 @@ export default class SearchBarComponent extends React.Component {
     this.setState({ query });
     if (query.length >= 2) {
       const http = new XMLHttpRequest();
-      const params = JSON.stringify({ query });
-      http.open('POST', '/api/v1/search', true);
+      http.open('GET', `${this.props.host}/search?query=${encodeURIComponent(query)}&mode=auto-suggest`, true);
```

Comment:
I'm curious, why do you decide to pass those configs from the server? Is there and advantage that I'm unaware of? I would have simply defined it in a constants module, then import the constants in the top of the component file and just retrieve the value with `constants.host`.